### PR TITLE
Hashmap improvements

### PR DESCRIPTION
**Pull request by aykevl**
Implement growing of hashmaps and setting the initial size. Necessary for `js.Func` support (#226), among others.

### CODE REVIEW

File: compiler/compiler.go [hunk 1]
```diff
@@ -1425,7 +1425,16 @@ func (c *Compiler) parseExpr(frame *Frame, expr ssa.Value) (llvm.Value, error) {
...
 		llvmKeySize := llvm.ConstInt(c.ctx.Int8Type(), keySize, false)
 		llvmValueSize := llvm.ConstInt(c.ctx.Int8Type(), valueSize, false)
-		hashmap := c.createRuntimeCall("hashmapMake", []llvm.Value{llvmKeySize, llvmValueSize}, "")
+		sizeHint := llvm.ConstInt(c.uintptrType, 8, false)
+		if expr.Reserve != nil {
+			sizeHint = c.getValue(frame, expr.Reserve)
+			var err error
+			sizeHint, err = c.parseConvert(expr.Reserve.Type(), types.Typ[types.Uintptr], sizeHint, expr.Pos())
+			if err != nil {
+				return llvm.Value{}, nil
```
Comment (aykevl) [inline compiler/compiler.go]: Oh no, that's a bug.

**Comment by aykevl** [comment]
Also fixed a leftover from https://github.com/tinygo-org/tinygo/pull/294 in the last commit (no functional change). Unrelated, but I thought I'd add it here as it is really trivial.

File: compiler/compiler.go (hunk 1 above)
Comment (aykevl) [reply compiler/compiler.go]: fixed.

**Comment by deadprogram** [comment]
OK, great, we need this. Now merging.

**Pull request merged**